t_clickhouse/test_clickhouse_logs.py

```diff
@@ -0,0 +1,422 @@
+            "id": "deproxy",
+            "type": "deproxy",
+            "addr": "${tempesta_ip}",
+            "port": "80",
+        }
+    ]
+
+    @staticmethod
+    def send_simple_request(deproxy_client, request=None, expected_status: str = "201") -> None:
+        """
+        The simple request with 200-code response
+        """
+        deproxy_client.send_request(
+            request=request or deproxy_client.create_request(method="GET", headers=[]),
+            expected_status_code=expected_status,
+            timeout=10,
+        )
+
+    def setUp(self):
+        super(TestClickhouseLogsBaseTest, self).setUp()
+        self.start_all_services(client=False)
+
+
+class TestClickhouseLogsBufferConfiguration(TestClickhouseLogsBaseTest):
+    tempesta = dict(
+        config="""
+            listen 80;
+            server ${server_ip}:8000;
+
+            mmap_log_buffer_size 4096;
+            access_log dmesg mmap mmap_host=${tfw_logger_clickhouse_host} mmap_log=${tfw_logger_daemon_log};
+        """
+    )
+
+    def test_mmap_buffer(self):
+        """
+        Check the buffer works fine
+        """
+        client = self.get_client("deproxy")
+        client.start()
+
+        self.send_simple_request(client)
+        self.assertEqual(self.loggers.dmesg.access_log_records_count(), 1)
+
+        for _ in range(4100):
+            self.send_simple_request(client)
+
+        self.assertEqual(self.loggers.clickhouse.access_log_records_count(), 4097)
+
+
+class TestClickhouseLogsOnly(TestClickhouseLogsBaseTest):
+    tempesta = dict(
+        config="""
+            listen 80;
+            server ${server_ip}:8000;
+
+            access_log mmap mmap_host=${tfw_logger_clickhouse_host} mmap_log=${tfw_logger_daemon_log};
+        """
+    )
+
+    def test_dmesg_toggled_off(self):
+        """
+        Toggle off the dmesg logs sending
+        """
+        client = self.get_client("deproxy")
+        client.start()
+
+        self.send_simple_request(client)
+        self.assertEqual(self.loggers.dmesg.access_log_records_count(), 0)
+        self.assertEqual(self.loggers.clickhouse.access_log_records_count(), 1)
+
+
+class TestClickhouseTFWLoggerFile(TestClickhouseLogsBaseTest):
+    tempesta = dict(
+        config="""
+            listen 80;
+            server ${server_ip}:8000;
+
+            access_log mmap mmap_host=${tfw_logger_clickhouse_host} mmap_log=${tfw_logger_daemon_log};
+        """
+    )
+
+    def test_twf_logger_file(self):
+        """
+        Check the content of tfw_logger daemon access log
+        """
+        client = self.get_client("deproxy")
+        client.start()
+
+        self.send_simple_request(client)
+        self.assertEqual(self.loggers.dmesg.access_log_records_count(), 0)
+
+        tempesta = self.get_tempesta()
+        tempesta.stop()
+
+        pattern = r".*Starting daemon.*Daemon started.*Stopping daemon.*Daemon stopped.*"
+        self.assertTrue(self.loggers.clickhouse.find(pattern))
```

Comment:
```suggestion
        self.send_simple_request(client)
        
        self.assertTrue(self.loggers.clickhouse.find("200")) # or other pattern for the requst
        
        self.loggers.dmesg.update()
        self.assertEqual(self.loggers.dmesg.access_log_records_count(), 0)

```
You must call `update` method for dmesg before `access_log_*` method or the log will be empty.